nmt_test.go

```diff
@@ -291,63 +291,176 @@ func TestNamespacedMerkleTree_ProveNamespace_Ranges_And_Verify(t *testing.T) {
 
 func TestIgnoreMaxNamespace(t *testing.T) {
 	var (
-		hash       = sha256.New()
-		nidSize    = 8
-		wantMinNID = []byte{0, 0, 0, 0, 0, 0, 0, 0}
-		wantMaxNID = []byte{0, 0, 0, 0, 0, 0, 0, 1}
-		maxNID     = []byte{0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF}
+		hash      = sha256.New()
+		nidSize   = 8
+		minNID    = []byte{0, 0, 0, 0, 0, 0, 0, 0}
+		secondNID = []byte{0, 0, 0, 0, 0, 0, 0, 1}
+		thirdNID  = []byte{0, 0, 0, 0, 0, 0, 0, 2}
+		maxNID    = []byte{0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF}
 	)
-	data := []namespace.Data{
-		namespace.PrefixedData8(append(wantMinNID, []byte("leaf_1")...)),
-		namespace.PrefixedData8(append(wantMaxNID, []byte("leaf_2")...)),
-		namespace.PrefixedData8(append(maxNID, []byte("leaf_3")...)),
-		namespace.PrefixedData8(append(maxNID, []byte("leaf_4")...)),
-	}
 
-	tree := New(hash, NamespaceIDSize(nidSize), IgnoreMaxNamespace(true))
-	treeNotIgnoredOpt := New(hash, NamespaceIDSize(nidSize), IgnoreMaxNamespace(false))
-	for _, d := range data {
-		if err := tree.Push(d); err != nil {
-			panic("unexpected error")
-		}
-		if err := treeNotIgnoredOpt.Push(d); err != nil {
-			panic("unexpected error")
-		}
-	}
-	gotDigest := tree.Root()
-	if !gotDigest.Min().Equal(wantMinNID) {
-		t.Fatalf("Unexpected root.Min() got: %x, want: %x", gotDigest.Min(), wantMinNID)
-	}
-	if !gotDigest.Max().Equal(wantMaxNID) {
-		t.Fatalf("Unexpected root.Max() got: %x, want: %x", gotDigest.Max(), wantMaxNID)
-	}
-	// if we explicitly do not ignore the max namespace, we expect it to show up in the root:
-	gotDigest2 := treeNotIgnoredOpt.Root()
-	if !gotDigest2.Max().Equal(maxNID) {
-		t.Fatalf("Unexpected root.Max() got: %x, want: %x", gotDigest2.Max(), maxNID)
+	tests := []struct {
+		name               string
+		ignoreMaxNamespace bool
+		pushData           []namespace.Data
+		wantRootMaxNID     namespace.ID
+	}{
+		{"single leaf with MaxNID (ignored)",
+			true,
+			[]namespace.Data{namespace.PrefixedData8(append(maxNID, []byte("leaf_1")...))},
+			maxNID,
+		},
+		{"single leaf with MaxNID (not ignored)",
+			false,
+			[]namespace.Data{namespace.PrefixedData8(append(maxNID, []byte("leaf_1")...))},
+			maxNID,
+		},
```

Comment:
The single leaf case is special in the sense as the only namespace won't be ignored. (We will have powers of 2 in LL anyways though).